### ISSUE TITLE
Feature Request: Sublocations

### ISSUE THREAD

**Issue by bartlibert** [grocy/grocy#481]
I'd like to be able to divide a location into several sublocations, e.g. several drawers in the refrigerator. I tried making each drawer a separate location, but then they all get printed on separate pages in the overview, while I'd like to have one location per page, but with subdivisions per sublocation.

E.g 

**Refrigerator** 
_Top drawer_ 
- 1kg spinach
- Spaghetti sauce
- ...

_Middle drawer_ 
- 500 g spinach
- lamb chops
- ...

I tried to do this with user defined field, but I did not find a way

**Comment by hannahswain** [comment]
I've solved this by using a structure in the location name, i.e. `Kitchen > Fridge`, or `Kitchen > Spice rack`. That keeps things grouped together nicely in the UI and on the print out.

**Comment by Stoicmaverick** [comment]
I posted the request for #671 , and this might be an option if nesting more than one location is possible. For instance my 100k resisters might be in "Workbench - Drawer 2 - Compartment Column B - Compartment Row 4, which might be displayed simply as "Workbench Drawer 2 Compartment B-4"

**Comment by t-gergely** [comment]
I need recursive locations, too, so I can migrate my inventory to Grocy from Piwigo. I have building/room/cupboard/shelf/side/box etc.

**Comment by jose1711** [comment]
Found http://www.twisterrob.net/project/inventory/ which seems to focus solely on home inventory. Maybe this can serve as a model/inspiration too.

**Comment by DieKatzchen** [comment]
The way the Magic Home Inventory works is kind of clever. All items are also locations, all locations are also items, so you can put anything inside anything. That way I can put the remote in a drawer, but I can also put the rechargeable batteries in the remote. Then you can search for an item and it will tell you where it is. If I could do this in Grocy it could also tell me when I put those batteries in and I could attach some notes like maybe for a universal remote I could put in the manual in case I ever need to reprogram it. Stuff like that.

**Comment by Saturo78** [comment]
A simple ERP warehouse logic would be perfect.
In the Location there are storage spaces.
100-01-01-01 stands  for Location 100-shelf 1-level 1- storage space 1
This Code could be printed as QR-Code an placed near the storage space. and can be used for the Transfer Transaction.

**Comment by berrnd** [comment]
@Saturo78 this is a about parent/child locations or something similar, definitely not about attaching barcodes to locations which is a duplicate of #221 (please keep it to one topic per request).

**Comment by mgh2** [comment]
Would love to have this as well. Looking to organize large locations into bins/containers, and sub locations would be great for this. 

Altho naming structure can, I can't see contents of multiple 'sublocations' simultaneously as I would imagine being able to do for a parent of nested locations.

**Comment by dgoethals** [comment]
Indeed it corresponds to a real need. For some family members, suffering from ADHD, organizing and finding items is a daily challenge. Nested locations would help them finding and storing items adequately, be it food or screws

**Comment by meistermolli** [comment]
Yes did some first tests with Grocy inside Homeassistant and this fits my needs.
i Will need Locations and Sub locations because the drop down is getting to big.
And want it to use as an storage location, too.

My need are 10 small Storage Containers with 33 Drawers that have 1/3 storage Places... its nearly 1000 Places for my small Items

I named it Container 1 / 1-A   with 1/3 2/3 3/3 till Container / 1 8-D
because I don't find anything if searching for a little Sensor or ESP.

Thanks
